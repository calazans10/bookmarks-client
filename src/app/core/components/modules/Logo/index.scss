@import 'styles/utilities';

.logo {
  flex: 1;

  a {
    font-size: em(22);
    font-weight: 400;
  }
}
