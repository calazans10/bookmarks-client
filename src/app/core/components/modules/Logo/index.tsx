import React from 'react';
import { Container, StyledLink } from './style';

type LogoProps = { to: string } & typeof defaultProps;

const defaultProps = {
  to: '/',
};

const Logo = ({ to }: LogoProps) => (
  <Container>
    <StyledLink to={to}>Bookmarks</StyledLink>
  </Container>
);

Logo.defaultProps = defaultProps;

export default Logo;
