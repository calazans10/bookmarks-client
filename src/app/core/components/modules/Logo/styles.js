import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { em } from 'polished';

export const Container = styled.div`
  flex: 1;
`;

export const StyledLink = styled(Link)`
  font-size: ${em('22px')};
  font-weight: 400;
`;
