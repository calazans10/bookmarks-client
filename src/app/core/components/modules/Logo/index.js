import React from 'react';
import PropTypes from 'prop-types';
import { Container, StyledLink } from './styles';

const Logo = ({ to }) => (
  <Container>
    <StyledLink to={to}>Bookmarks</StyledLink>
  </Container>
);

Logo.propTypes = {
  to: PropTypes.string,
};

Logo.defaultProps = {
  to: '/',
};

export default Logo;
