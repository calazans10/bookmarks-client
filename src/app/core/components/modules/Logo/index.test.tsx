import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Logo from '.';

describe('Logo', () => {
  it('renders without crashing', () => {
    const { container } = render(<Logo to="/" />, { wrapper: MemoryRouter });
    expect(container.firstChild).toMatchSnapshot();
  });
});
