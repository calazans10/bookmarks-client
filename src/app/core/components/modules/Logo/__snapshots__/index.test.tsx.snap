// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders without crashing 1`] = `
<div
  class="sc-AxjAm daLWxe"
>
  <a
    class="sc-AxirZ gHWAZY"
    href="/"
  >
    Bookmarks
  </a>
</div>
`;
